hs_modflow_modelinstance/tests/test_modflow_modelinstance_metadata.py

```diff
@@ -0,0 +1,1241 @@
+        self.res.metadata.create_element('ModelOutput', includes_output=True)
+        self.res.metadata.create_element('ExecutedBy', model_name=self.resGenModelProgram.short_id)
+        self.res.metadata.create_element('StudyArea',
+                                         totalLength=1111,
+                                         totalWidth=2222,
+                                         maximumElevation=3333,
+                                         minimumElevation=4444)
+        self.res.metadata.create_element('GridDimensions',
+                                         numberOfLayers=5555,
+                                         typeOfRows='Irregular',
+                                         numberOfRows=6666,
+                                         typeOfColumns='Regular',
+                                         numberOfColumns=7777)
+        self.res.metadata.create_element('StressPeriod',
+                                         stressPeriodType='Steady and Transient',
+                                         steadyStateValue=8888,
+                                         transientStateValueType='Monthly',
+                                         transientStateValue=9999)
+        self.res.metadata.create_element('GroundwaterFlow',
+                                         flowPackage='LPF',
+                                         flowParameter='Hydraulic Conductivity')
+        self.res.metadata.create_element('BoundaryCondition',
+                                         specified_head_boundary_packages=['CHD', 'FHB'],
+                                         specified_flux_boundary_packages=['FHB', 'WEL'],
+                                         head_dependent_flux_boundary_packages=['RIV', 'MNW1'])
+        self.res.metadata.create_element('ModelCalibration',
+                                         calibratedParameter='test parameter',
+                                         observationType='test observation type',
+                                         observationProcessPackage='GBOB',
+                                         calibrationMethod='test calibration method')
+        self.res.metadata.create_element('ModelInput',
+                                         inputType='test input type',
+                                         inputSourceName='test source name',
+                                         inputSourceURL='http://www.test.com')
+        self.res.metadata.create_element('GeneralElements',
+                                         modelParameter='test model parameter',
+                                         modelSolver='SIP',
+                                         output_control_package=['HYD', 'OC'],
+                                         subsidencePackage='SWT')
+        xml_doc = self.res.metadata.get_xml()
+        # check to see if the specific metadata are in the xml doc
+        self.assertTrue('1111' in xml_doc)

```

Comment:
This is probably not the correct way to test the xml representation of each of the metadata element in the xml document. However, none of the resource types including generic resource has not implemented the proper way to test the generated xml document. I hope to do that someday.